search/elastic.py

```diff
@@ -292,151 +351,121 @@ def _check_mappings(self, doc_type, body):
+        except exceptions.ElasticsearchException as ex:
+            log.exception("Error during ES bulk operation.")
             raise
 
-    def remove(self, doc_type, doc_ids, **kwargs):
-        """ Implements call to remove the documents from the index """
+    def remove(self, doc_ids, **kwargs):
+        """
+        Implements call to remove the documents from the index
+        """
 
         try:
-            # ignore is flagged as an unexpected-keyword-arg; ES python client documents that it can be used
-            # pylint: disable=unexpected-keyword-arg
             actions = []
             for doc_id in doc_ids:
-                log.debug("Removing document of type %s and index %s", doc_type, doc_id)  # lint-amnesty, pylint: disable=unicode-format-string
+                log.debug("Removing document with id %s", doc_id)
                 action = {
-                    '_op_type': 'delete',
+                    "_op_type": "delete",
                     "_index": self.index_name,
-                    "_type": doc_type,
                     "_id": doc_id
                 }
                 actions.append(action)
             bulk(self._es, actions, **kwargs)
         except BulkIndexError as ex:
-            valid_errors = [error for error in ex.errors if error['delete']['status'] != 404]
+            valid_errors = [error for error in ex.errors if error["delete"]["status"] != 404]
 
             if valid_errors:
-                log.exception("An error occurred while removing documents from the index.")
+                log.exception("An error occurred while removing documents from the index: %r", valid_errors)
                 raise
 
-    # A few disabled pylint violations here:
-    # This procedure takes each of the possible input parameters and builds the query with each argument
-    # I tried doing this in separate steps, but IMO it makes it more difficult to follow instead of less
-    # So, reasoning:
-    #
-    #   too-many-arguments: We have all these different parameters to which we
-    #       wish to pay attention, it makes more sense to have them listed here
-    #       instead of burying them within kwargs
-    #
-    #   too-many-locals: I think this counts all the arguments as well, but
-    #       there are some local variables used herein that are there for transient
-    #       purposes and actually promote the ease of understanding
-    #
-    #   too-many-branches: There's a lot of logic on the 'if I have this
-    #       optional argument then...'. Reasoning goes back to its easier to read
-    #       the (somewhat linear) flow rather than to jump up to other locations in code
     def search(self,
                query_string=None,
                field_dictionary=None,
                filter_dictionary=None,
                exclude_dictionary=None,
-               facet_terms=None,
+               agg_terms=None,
```

Comment:
Nit: I'm not a fan of abbreviations in variable names, so I'd prefer to see this as `aggregate_terms`/`aggregation_terms` (?).

Actually, how specific do we need/want to be? Could these be referred to simply as `terms` or is that too broad?

Reply:
Done. I chose  `aggregation_terms` abbreviation